### PR TITLE
Get spec_interval string from CarbonInterval

### LINKED ISSUE #525: Ability to get interval_spec from CarbonInterval
This is a feature request that I am happy to implement and send a Pull request with, but before that I'd like to discuss it.

I am currently in a need of getting `interval_spec` from `CarbonInterval` object. I imagine it could be achieved by implementing `spec` method on the `CarbonInterval` class. Do you see any potential issues with this?

### PR DESCRIPTION
**Pull request by martinssipenko**
Fixes #525

New version of https://github.com/briannesbitt/Carbon/pull/526 that removes updates to `composer.json`

### CODE REVIEW

**Comment by dereuromark** [comment]
I still think it would be better to use __toString() here as I did in https://github.com/cakephp/chronos/pull/59 respectively.

**Comment by martinssipenko** [comment]
@GrahamCampbell Can you help me with failing StyleCI, I'm not sure I understand the issue.

Thanks!

File: src/Carbon/CarbonInterval.php [hunk 1]
```diff
@@ -519,4 +519,39 @@ public function add(DateInterval $interval)
...
+        $specString = static::PERIOD_PREFIX;
+
+        foreach ($date as $key => $value) {
+            $specString .= $value . $key;
+        }
+
+        if (count($time) > 0) {
+            $specString .= static::PERIOD_TIME_PREFIX;
+            foreach ($time as $key => $value) {
+            $specString .= $value . $key;
```
Comment (dereuromark) [inline src/Carbon/CarbonInterval.php]: this line is shown as invalid: https://styleci.io/analyses/q5v6Pz

**Comment by martinssipenko** [comment]
Sorry, my bad, was looking at wrong line.

**Comment by nicholas-eden** [comment]
This results in a lot of duplication of code in the constructor.  Maybe have a buildSpec method that accepts $years, $months, etc.  Then have both the constructor and spec method call it.

**Pull request merged**